File: days/20220315/main.py
# %%
## 1249. Minimum Remove to Make Valid Parentheses
## author: Greysuki
class Solution:
    def minRemoveToMakeValid(self, s: str) -> str:
        c = 0
        stack = []
        split_s = list(s)
        while c < len(split_s):
            if split_s[c] == "(":
                stack.append(c)
            elif split_s[c] == ")":
                if len(stack) > 0:
                    stack.pop()
                else:
                    split_s[c] = ""
            c += 1

        while stack:
            split_s[stack.pop()] = ""

        return "".join(split_s)


# Your Solution object will be instantiated and called as such:
obj = Solution()
print(obj.minRemoveToMakeValid("lee(t(c)o)de)"))

File: days/20220315/main.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    string minRemoveToMakeValid(string s)
    {
        int c = 0;
        vector<string> st;
        string new_s = "";
        cout << s.length() << endl;
        st.push_back("TTT");
        while (c < s.length())
        {
            cout << new_s << endl;
            cout << s[c] << endl;
            if (s[c] == '(')
            {
                // stack.push_back(c);
            }
            // else if (s[c] == ')')
            // {
            //     if (stack.size() > 0)
            //     {
            //         stack.pop_back();
            //     }
            //     else
            //     {
            //         continue;
            //     }
            // }

            new_s += s[c];
            c += 1;
        }

        return new_s;
    }
};

int main()
{
    Solution sol = Solution();
    auto ret = sol.minRemoveToMakeValid("lee(t(c)o)de)");
    cout << ret << std::endl;
    cout << "Hello world" << endl;

    return 0;
}
